object-manipulation-2/src/defaults.js

```diff
@@ -0,0 +1,32 @@
+// /* eslint-disable no-unused-vars */
+
+/*
+Assigns properties of source object to the target object for all properties that are not defined on target.
+*/
+
+const defaults = (target, source) => {
+  const targetProp = [];
+  const sourceProp = [];
+
+  for (const property in target) {
+    targetProp.push(property);
+  }
+
+  for (const property in source) {
+    sourceProp.push(property);
+  }
+
+  let counter = 0;
+  for (let i = 0; i < sourceProp.length; i++) {
+    for (let j = 0; j < targetProp.length; j++) {
+      if (sourceProp[i] !== targetProp[j]) {
+        counter++;
+      }
+    }
+    if (counter === targetProp.length) {
+      target[sourceProp[i]] = source[sourceProp[i]];
+    }
+    counter = 0;
+  }
+  return target;
```

Comment:
This solution can be simplified quite a bit. Try:
1. looping through source keys
1. if the target is undefined at a source key
1. assign source at key to target at key

Reply:
revised!